pkg/operator/controller/controller_dns_daemonset.go

```diff
@@ -230,6 +230,58 @@ func daemonsetConfigChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.
 	return true, updated
 }
 
+// cmpConfigMapVolumeSource compares two configmap volume source values and
+// returns a Boolean indicating whether they are equal.
+func cmpConfigMapVolumeSource(a, b corev1.ConfigMapVolumeSource) bool {
+	if a.LocalObjectReference != b.LocalObjectReference {
+		return false
+	}
+	if !cmp.Equal(a.Items, b.Items, cmpopts.EquateEmpty()) {
+		return false
+	}
+	aDefaultMode := int32(420)
+	if a.DefaultMode != nil {
+		aDefaultMode = *a.DefaultMode
+	}
+	bDefaultMode := int32(420)
+	if b.DefaultMode != nil {
+		bDefaultMode = *b.DefaultMode
+	}
+	if aDefaultMode != bDefaultMode {
+		return false
+	}
+	if !cmp.Equal(a.Optional, b.Optional, cmpopts.EquateEmpty()) {
+		return false
+	}
+	return true
```

Comment:
nit: could condense lines 253-256 to `return cmp.Equal(a.Optional, b.Optional, cmpopts.EquateEmpty())`

Reply:
Stylistically, I prefer to follow the pattern consistently:
```go
	if thing1 doesn't match {
		return false
	}
	if thing2 doesn't match {
		return false
	}
	// ...
	if thingn doesn't match {
		return false
	}
	return true
```
There is nothing special about thingn (in concrete terms, the `Optional` field), so I prefer not to deviate from the established pattern.  Does that seem reasonable?

Reply:
yes, it does. I see what you mean now!